pkg/stdcopy/stdcopy_test.go

```diff
@@ -85,6 +85,24 @@ func TestWriteDoesNotReturnNegativeWrittenBytes(t *testing.T) {
 	}
 }
 
+func TestStdCopyWriteAndRead(t *testing.T) {
+	buffer := new(bytes.Buffer)
+	stdOutBytes := []byte(strings.Repeat("o", startingBufLen))
+	dstOut := NewStdWriter(buffer, Stdout)
+	dstOut.Write(stdOutBytes)
+	stdErrBytes := []byte(strings.Repeat("e", startingBufLen))
+	dstErr := NewStdWriter(buffer, Stderr)
+	dstErr.Write(stdErrBytes)
+	written, err := StdCopy(dstOut, dstErr, bytes.NewReader(buffer.Bytes()))
```

Comment:
Upon a second thought I think it's actually a good idea for clarity to not reuse dstOut and dstErr here. I've updated the code to use buffer as you suggested and switched io.Writers to ioutil.Discard in StdCopy call. No infinite loop now.